### PR TITLE
Feature/division builder

### PR DESCRIPTION
**Pull request by markspanbroek**
Added a division builder to simplify the creation of division protocols.

### CODE REVIEW

**Comment by Rhaea** [comment]
Not sure if we want a single builder for just division. Maybe more something like AdvancedNumericBuilder. Not entirely sure what else to put in that right now, but I'm sure more features will emerge. So do a renaming for now I guess, and I, for one, will be happy. 
Like the change with the abstract class as well! Makes sense

**Comment by markspanbroek** [comment]
Good suggestion @Rhaea, I've renamed the class.

File: src/main/java/dk/alexandra/fresco/lib/helper/builder/AdvancedNumericBuilder.java [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    private DivisionFactory divisionFactory;
+    private SIntFactory intFactory;
+
+    public AdvancedNumericBuilder(DivisionFactory divisionFactory, SIntFactory intFactory) {
+        this.divisionFactory = divisionFactory;
+        this.intFactory = intFactory;
+    }
+
+    public SInt div(SInt p, int maxBitLengthP, SInt q, int maxBitLengthQ) {
```
Comment (Rhaea) [inline src/main/java/dk/alexandra/fresco/lib/helper/builder/AdvancedNumericBuilder.java]: Oh yea.. two things: javadoc missing. One does not know if p/q or q/p. And maybe call them divident and divisor. 
And second thing: We would like methods for all the possible functionality that the division factory provides. So also division by known divisor and division where the remainder is returned as well etc.

**Comment by markspanbroek** [comment]
Ok, I've added javadoc and included the other operations from the division factory. Could you please merge this pull request @Rhaea? Then I can create the next pull request (an improved division algorithm) which depends on the changes in this one.

**Comment by Rhaea** [comment]
The policy is to have at least two reviewers look at a pull request before merging. This should happen soon though - I pinged Peter today.

**Pull request merged**